heron/common/src/python/constants.py

```diff
@@ -11,20 +11,191 @@
+TOPOLOGY_MAX_SPOUT_PENDING = "topology.max.spout.pending"
+# A list of task hooks that are automatically added to every spout and bolt in the topology.
+TOPOLOGY_AUTO_TASK_HOOKS = "topology.auto.task.hooks"
+# The serialization class that is used to serialize/deserialize tuples
+TOPOLOGY_SERIALIZER_CLASSNAME = "topology.serializer.classname"
+# How many executors to spawn for ackers.
+TOPOLOGY_ENABLE_ACKING = "topology.acking"
+# Number of cpu cores per container to be reserved for this topology.
+TOPOLOGY_CONTAINER_CPU_REQUESTED = "topology.container.cpu"
+# Amount of ram per container to be reserved for this topology, in bytes.
 TOPOLOGY_CONTAINER_RAM_REQUESTED = "topology.container.ram"
+# Amount of disk per container to be reserved for this topology, in bytes.
 TOPOLOGY_CONTAINER_DISK_REQUESTED = "topology.container.disk"
+# Hint for max amount of ram per container to be reserved for this topology, in bytes.
+TOPOLOGY_CONTAINER_MAX_CPU_HINT = "topology.container.max.cpu.hint"
+# Hint for max amount of disk per container to be reserved for this topology, in bytes.
+TOPOLOGY_CONTAINER_MAX_DISK_HINT = "topology.container.max.disk.hint"
+# Padding percentage for this container.
+TOPOLOGY_CONTAINER_PADDING_PERCENTAGE = "topology.container.padding.percentage"
+
+# Per component ram requirement.
 TOPOLOGY_COMPONENT_RAMMAP = "topology.component.rammap"
-TOPOLOGY_STMGRS = "topology.stmgrs"
-TOPOLOGY_WORKER_CHILDOPTS = "topology.worker.childopts"
+# Name of the topology, automatically set by Heron when the topology is submitted.
+TOPOLOGY_NAME = "topology.name"
+# Name of the team which owns this topology.
+TOPOLOGY_TEAM_NAME = "topology.team.name"
+# Email of the team which owns this topology.
+TOPOLOGY_TEAM_EMAIL = "topology.team.email"
+# Cap ticket (if filed) for the topology. If the topology is in prod, this has to be set or it
+# cannot be deployed.
+TOPOLOGY_CAP_TICKET = "topology.cap.ticket"
+# Project name of the topology, to help us with tagging which topologies are part of which project.
+TOPOLOGY_PROJECT_NAME = "topology.project.name"
+# Any user defined classpath that needs to be passed to instances should be set in to config
+# through this key.
 TOPOLOGY_ADDITIONAL_CLASSPATH = "topology.additional.classpath"
-RAM_FOR_STMGR = 1 * GB
-DEFAULT_RAM_FOR_INSTANCE = 1 * GB
-DEFAULT_DISK_PADDING_PER_CONTAINER = 12 * GB
+
+
+##### Constants for system configuration #####
+
+# The relative path to the logging directory
+HERON_LOGGING_DIRECTORY = 'heron.logging.directory'
+# The maximum log file size in MB
+HERON_LOGGING_MAXIMUM_SIZE_MB = "heron.logging.maximum.size.mb"
+# The maximum number of log files
+HERON_LOGGING_MAXIMUM_FILES = "heron.logging.maximum.files"
+# The threshold level to log error
+HERON_LOGGING_ERR_THRESHOLD = "heron.logging.err.threshold"
+# The interval in seconds to get and reset the system metrics.
+HERON_METRICS_EXPORT_INTERVAL_SEC = "heron.metrics.export.interval.sec"
+
```

Comment:
Can you add a huge separator comment between big groups of configuration? For example, before instance specific configuration - can you add a huge comment so that when we read the file we can get to specific group quickly?